test/surgex/changeset/changeset_test.exs

```diff
@@ -2,32 +2,97 @@ defmodule Surgex.ChangesetTest do
+        end
+      end
+    end
+
+    def changeset(schema, params) do
+      schema
+      |> cast(params, [:name])
+      |> cast_embed(:address, with: &address_changeset/2)
+    end
+
+    defp address_changeset(schema, params) do
+      schema
+      |> cast(params, [:city, :street])
+      |> validate_required([:city])
+      |> cast_embed(:country, with: &country_changeset/2)
+    end
+
+    defp country_changeset(schema, params) do
+      schema
+      |> cast(params, [:code])
+      |> validate_inclusion(:code, ["us", "ca", "pl"])
+    end
+  end
+
   test "changeset with all possible errors" do
-    assert Changeset.build_errors_document(%{
-             errors: [
-               taken_field: {"has already been taken", nil},
-               required_field: {nil, [validation: :required]},
-               invalid_field: {nil, nil},
-               invalid_type_field: {nil, [validation: :type]}
-             ]
-           }) == %Jabbax.Document{
+    changeset =
+      %TestSchema{}
+      |> change(%{})
+      |> add_error(:taken_field, "has already been taken")
+      |> add_error(:required_field, "", validation: :required)
+      |> add_error(:invalid_type_field, "", validation: :type)
+      |> add_error(:invalid_field, "")
+
+    assert Changeset.build_errors_document(changeset) == %Jabbax.Document{
              errors: [
                %Jabbax.Document.Error{
-                 code: "taken",
-                 source: %Jabbax.Document.ErrorSource{pointer: "/data/attributes/taken_field"}
+                 code: "invalid",
+                 source: %Jabbax.Document.ErrorSource{pointer: "/data/attributes/invalid_field"}
+               },
+               %Jabbax.Document.Error{
+                 code: "invalid_type",
+                 source: %Jabbax.Document.ErrorSource{
+                   pointer: "/data/attributes/invalid_type_field"
+                 }
                },
                %Jabbax.Document.Error{
                  code: "required",
                  source: %Jabbax.Document.ErrorSource{pointer: "/data/attributes/required_field"}
                },
                %Jabbax.Document.Error{
-                 code: "invalid",
-                 source: %Jabbax.Document.ErrorSource{pointer: "/data/attributes/invalid_field"}
+                 code: "taken",
+                 source: %Jabbax.Document.ErrorSource{pointer: "/data/attributes/taken_field"}
+               }
+             ]
+           }
+  end
+
+  test "changeset with nested errors" do
+    changeset =
+      TestSchema.changeset(%TestSchema{}, %{
+        name: "test",
+        address: %{street: "Main", country: %{code: "es"}}
+      })
+
+    assert Changeset.build_errors_document(changeset) == %Jabbax.Document{
+             errors: [
+               %Jabbax.Document.Error{
+                 code: "required",
+                 source: %Jabbax.Document.ErrorSource{
+                   pointer: "/relationships/address/data/attributes/city"
```

Comment:
I don't think this will be a relationship in a JSON:API document - an embedded schema rather maps to an attribute with an `object` value. You shouldn't send [resource objects](https://jsonapi.org/format/#document-resource-objects) through relationships, rather [resource identifier objects](https://jsonapi.org/format/#document-resource-identifier-objects). So I think this should rather be:

```suggestion
                   pointer: "/attributes/address/city"
```

Reply:
This would be addressed by https://github.com/surgeventures/jabbax/pull/8